packages/salesforcedx-vscode-core/src/orgBrowser/metadataCmp.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as fs from 'fs';
+import * as path from 'path';
+import { Observable } from 'rxjs/Observable';
+import { isNullOrUndefined } from 'util';
+import * as vscode from 'vscode';
+import {
+  EmptyParametersGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from '../commands';
+import { nls } from '../messages';
+import { notificationService } from '../notifications';
+import { taskViewService } from '../statuses';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath, hasRootWorkspace, OrgAuthInfo } from '../util';
+
+export class ForceListMetadataExecutor extends SfdxCommandletExecutor<string> {
+  private metadataType: string;
+  private outputPath: string;
+  private defaultUsernameOrAlias: string;
+
+  public constructor(
+    metadataType: string,
+    outputPath: string,
+    defaultUsernameOrAlias: string
+  ) {
+    super();
+    this.metadataType = metadataType;
+    this.outputPath = outputPath;
+    this.defaultUsernameOrAlias = defaultUsernameOrAlias;
+  }
+
+  public build(data: {}): Command {
+    let builder = new SfdxCommandBuilder()
+      .withArg('force:mdapi:listmetadata')
+      .withFlag('-m', this.metadataType)
+      .withFlag('-u', this.defaultUsernameOrAlias)
+      .withFlag('-f', this.outputPath)
```

Comment:
Are we using output files instead of directly passing the data to the tree provider because we're planning on having an org's component tree persist between vs code sessions?

cc: @lcampos

Reply:
@brpowell yes, we eventually want to allow users to move across orgs and having all the info in memory is not a good option. We'll store the data under `.sfdx` and will access it and update it based on the user's actions via the UI.